From: Ben Hutchings <benh@debian.org>
Date: Thu, 31 Dec 2020 19:18:15 +0100
Subject: Revert "seq_buf: Avoid type mismatch for seq_buf_init"
Forwarded: not-needed

Temporarily revert commit 4cf97fe9cd8010c33748ab92270c1e3cef2dd655 to
avoid an ABI change.

---
--- b/include/linux/seq_buf.h
+++ a/include/linux/seq_buf.h
@@ -29,7 +29,7 @@
 }
 
 static inline void
+seq_buf_init(struct seq_buf *s, unsigned char *buf, unsigned int size)
-seq_buf_init(struct seq_buf *s, char *buf, unsigned int size)
 {
 	s->buffer = buf;
 	s->size = size;
reverted:
--- b/include/linux/trace_seq.h
+++ a/include/linux/trace_seq.h
@@ -11,7 +11,7 @@
  */
 
 struct trace_seq {
+	unsigned char		buffer[PAGE_SIZE];
-	char			buffer[PAGE_SIZE];
 	struct seq_buf		seq;
 	int			full;
 };
@@ -50,7 +50,7 @@
  * that is about to be written to and then return the result
  * of that write.
  */
+static inline unsigned char *
-static inline char *
 trace_seq_buffer_ptr(struct trace_seq *s)
 {
 	return s->buffer + seq_buf_used(&s->seq);
